### PR TITLE
New Resources, Full Json Implementation for Activities

### PR DESCRIPTION
**Pull request by wchen-oxy**
-Updated format for each entry in Nutrition/Exercise
-Updated each Json Read function with code to catch Null exceptions
-Passed Data successfully to detailed activity using parceable
-Added basic detailed activity layout
-Added new image resources for icon and main activities

### CODE REVIEW

**Comment by wchen-oxy** [comment]
I am aware that Dan's and my item format have similar changes. I had already made my changes when he pull requested. Not sure what to do but I'll pull request for now since the bulk of my changes are distinct from his. @justinnhli @crzeller11

**Comment by wchen-oxy** [comment]
What should I do for the failed Travis CI build? It says my png resource failed because it doesn't contain the signature png format. Why should that matter if the file itself ends in .png?

**Comment by justinnhli** [comment]
1. What's the difference between this PR and #13?

2. It might be that it ends in .png but is actually a jpeg file.

**Comment by wchen-oxy** [comment]
#13 has only itemformat changed for nutrition. I have that changed as well using different variables. However, everything else I have listed under the updates are different than #13.

So to be more specific:
Same:
-Updated format for each entry in Nutrition

Different
-Updated format for each entry in Exercise
-Updated each Json Read function with code to catch Null exceptions
-Passed Data successfully to detailed activity using parceable
-Added basic detailed activity layout
-Added new image resources for icon and main activities

I don't think that its a jpeg file. I exported it as a  png through photoshop. The metadata says it's a png as well. I could reupload it again though.

**Comment by blackjamx5** [comment]
I looked at the metadata for the pngs I uploaded and they don't appear to be jpegs

**Comment by justinnhli** [comment]
It's a JPEG:

screen shot 2018-04-15 at 14 36 00

Open it in Preview/Photoshop or whatever and force save it as a png.

**Comment by justinnhli** [comment]
Re: the differences from #13 - I think you and @Danhuth need to figure out what the right thing to do is. I believe the changes currently conflict, which means at least parts of one of these PRs will be overwritten. Figure that out and tag me again.

**Comment by wchen-oxy** [comment]
Sorry my mistake. I checked using preview but not using the detailed info. I've uploaded the new png and will figure out the changes with Dan

**Comment by wchen-oxy** [comment]
I've talked it over with Dan and we're going to use my code. I updated my variable names so that they fit his documentation. I've also updated the code with Justin's suggestions to Dan.
@justinnhli @crzeller11 

The code should be ready for merging.

File: app/build.gradle
Comment (justinnhli) [inline app/build.gradle]: use `implementation` instead of `compile`

File: app/src/main/java/practicum/getfitla_v3/ExerciseDetailActivity.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/ExerciseDetailActivity.java]: delete unused code

File: app/src/main/java/practicum/getfitla_v3/ExerciseDetailActivity.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/ExerciseDetailActivity.java]: comment no longer accurage

File: app/src/main/java/practicum/getfitla_v3/ExerciseDetailActivity.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/ExerciseDetailActivity.java]: what is this variable?

File: app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java]: reformat this line and remove the unused parameters

File: app/src/main/java/practicum/getfitla_v3/Exercise.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/Exercise.java]: delete

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/Nutrition.java]: EXERCISE_SHEET_URL in Nutrition.java?

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/Nutrition.java]: variable naming

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/Nutrition.java]: delete

File: app/src/main/java/practicum/getfitla_v3/Exercise.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/Exercise.java]: This gets repeated a lot - can you put it in a function instead?

File: app/src/main/java/practicum/getfitla_v3/ExerciseListAdapter.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/ExerciseListAdapter.java]: delete

File: app/src/main/java/practicum/getfitla_v3/NutritionListAdapter.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/NutritionListAdapter.java]: delete

**Review by justinnhli — CHANGES_REQUESTED** [review]
In addition to the in-line comments, also clean up the vertical whitespace - lots of places have multiple blank lines that make the code difficult to read.

File: app/build.gradle
Comment (crzeller11) [reply app/build.gradle]: ^^^avoid so many warnings with gradle build

File: app/src/main/java/practicum/getfitla_v3/ExerciseItemFormat.java
Comment (crzeller11) [inline app/src/main/java/practicum/getfitla_v3/ExerciseItemFormat.java:29]: I made this comment on Dan's code, but I'm not sure why you need all these methods. Can't you just get the objects attributes by doing: 
```
int id = exampleObject.id;
```
I guess I don't see the point in writing another method to do the same thing

File: app/src/main/java/practicum/getfitla_v3/ExerciseListAdapter.java
Comment (crzeller11) [inline app/src/main/java/practicum/getfitla_v3/ExerciseListAdapter.java]: This is probably fine, but I usually just write 
```
return fullList.size();
``` 
I don't think you need to catch the exception when it's 0

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (crzeller11) [inline app/src/main/java/practicum/getfitla_v3/Nutrition.java]: So you can put both the JSON process/object creation together with the code that you actually render the view with, but it would be cleaner to have one class where you read in all this JSON shit and write a method that throws it all in an ArrayList, and then have  separate class to do onCreate

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (crzeller11) [inline app/src/main/java/practicum/getfitla_v3/Nutrition.java]: You repeat this line a lot, write a method to make it cleaner.

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (crzeller11) [reply app/src/main/java/practicum/getfitla_v3/Nutrition.java]: Yes, also confused by this.

File: app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java
Comment (crzeller11) [inline app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java]: you only need passedname and passedprocess to set up your text view below, you can delete everything else.

File: app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java
Comment (crzeller11) [inline app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java]: Only declare what you need to show in the view. The only variables you need are passedname and passedprocess

File: app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java
Comment (crzeller11) [inline app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:37]: Again, I think these methods are unnecessary

**Review by crzeller11 — CHANGES_REQUESTED** [review]

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (wchen-oxy) [reply app/src/main/java/practicum/getfitla_v3/Nutrition.java]: I blanked out the things associated with rating and price because I had yet to add rating and prices to the google sheets and layouts. Can I keep it for now?

File: app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java
Comment (wchen-oxy) [reply app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java]: @Danhuth Can u give me the equivalent NUTRITION_SHEET_URL

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (justinnhli) [reply app/src/main/java/practicum/getfitla_v3/Nutrition.java]: I would add it to the Google Sheet as empty columns, then uncomment them here.

File: app/src/main/java/practicum/getfitla_v3/ExerciseItemFormat.java
Comment (wchen-oxy) [reply app/src/main/java/practicum/getfitla_v3/ExerciseItemFormat.java:29]: I figured out what word I meant to say when I tried to explain why I want to keep my strings private: encapsulation. Keeping variables private ensures that I don't accidentally override anything.
http://javarevisited.blogspot.com/2012/03/what-is-encapsulation-in-java-and-oops.html

@crzeller11

**Comment by wchen-oxy** [comment]
@justinnhli @crzeller11 

Everything should be fixed now. If there are unused passed variables in the detail activity, that's only because we haven't used it in the layouts yet. Joey/Me will be taking care of as soon as the rest are merged onto the main copy. 

I kept the extra get methods because of encapsulation.

**Comment by justinnhli** [comment]
There are still unfixed comments. Scroll back up this page to see Chloe and my notes on your code.

File: app/src/main/res/values/strings.xml
Comment (justinnhli) [inline app/src/main/res/values/strings.xml]: I'm assuming this is left over from a merge conflict resolution

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/Nutrition.java]: I have so many questions about this above section of code. Why is the arraylist named `Temp` (with a capital T, even)? What is `x`? Why do you keep adding `x` to `Temp` then getting the first thing?

Comment (wchen-oxy) [reply app/src/main/java/practicum/getfitla_v3/Nutrition.java]: I use these both in for loop, where each loop contains a counter starting at zero. This was a dumb workaround to return counter (int) and to return the value (string) from the add() method. They way I had written it wasn't letting me return two things and/or modify the counter when the add() was called. I'm not sure how to call the add() method, modify a counter that was initialized in the for loop and return a value found in the Json.

Comment (justinnhli) [reply app/src/main/java/practicum/getfitla_v3/Nutrition.java]: Hm. Can you do something like (untested):

```java
public String getFromRow(JSONObject row, int counter) {
    try {
        //check if valid
        return row.getJSONArray("c").getJSONObject(counter).optString("v");
    } catch (JSONException ex) {
        //exception for null values
        return "";
    }
}
```

Then in `processJson`:

```java
//initialize secondary counter to control for the index
int counter = 0;

String name = getFromRow(row, counter);
counter++;
String prepTime = getFromRow(row, counter);
counter++;
String servingSize = getFromRow(row, counter);
counter++;
String calories = getFromRow(row, counter);
counter++;
String equipment = getFromRow(row, counter);
counter++;
String ingredients = getFromRow(row, counter);
counter++;
String process = getFromRow(row, counter);
counter++;
String shortdesc = getFromRow(row, counter);
counter++;

// Temporary fix until we have images
int image = R.drawable.nutrition;
```

Comment (justinnhli) [reply app/src/main/java/practicum/getfitla_v3/Nutrition.java]: Or if you want to be fancy, the second part can be:

```java
//initialize secondary counter to control for the index
int counter = 0;

String name = getFromRow(row, counter++);
String prepTime = getFromRow(row, counter++);
String servingSize = getFromRow(row, counter++);
String calories = getFromRow(row, counter++);
String equipment = getFromRow(row, counter++);
String ingredients = getFromRow(row, counter++);
String process = getFromRow(row, counter++);
String shortdesc = getFromRow(row, counter++);

// Temporary fix until we have images
int image = R.drawable.nutrition;
```

The first version is probably easier to understand.

Comment (justinnhli) [reply app/src/main/java/practicum/getfitla_v3/Nutrition.java]: I take everything back. This is the better way:

```java
NutritionItemFormat nutrition = new NutritionItemFormat(
    row_id,
    getFromRow(row, 0), // name
    getFromRow(row, 7), // shortdesc
    image,
    getFromRow(row, 1), // prepTime
    getFromRow(row, 2), // servingSize
    getFromRow(row, 3), // calories
    getFromRow(row, 4), // equipment
    getFromRow(row, 6), // process
    getFromRow(row, 5), // ingredients
    rating,
    price
);
```

File: app/src/main/java/practicum/getfitla_v3/ExerciseItemFormat.java
Comment (crzeller11) [reply app/src/main/java/practicum/getfitla_v3/ExerciseItemFormat.java:29]: Ah ok, I wasn't familiar with this, but this is fine!

File: app/build.gradle
Comment (justinnhli) [reply app/build.gradle]: This has yet to be fixed.

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (crzeller11) [reply app/src/main/java/practicum/getfitla_v3/Nutrition.java]: ^I agree with Justin's code, it looks much cleaner and it's much easier to see what's going on

File: app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java
Comment (crzeller11) [inline app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java:17]: This whole section could be made simpler by doing this for each TextView you're setting up:
```
TextView title = (TextView) findViewById(R.id.recipe_title);
title.setText(CurItem.getName());
```

**Review by crzeller11 — CHANGES_REQUESTED** [review]

File: app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java
Comment (justinnhli) [reply app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java:17]: Or even:

```java
( (TextView) findViewById(R.id.recipe_title) ).setText(CurItem.getName());
```

File: app/build.gradle
Comment (wchen-oxy) [reply app/build.gradle]: Gosh, yes you're right. I didn't commit the gradle file since we established awhile ago to not upload builds generated by android studio. My mistake.

File: app/src/main/java/practicum/getfitla_v3/Nutrition.java
Comment (wchen-oxy) [reply app/src/main/java/practicum/getfitla_v3/Nutrition.java]: I'll give it a shot!

**Comment by crzeller11** [comment]
@justinnhli looks fine to me, but I'll wait for you to look before I merge

File: app/src/main/java/practicum/getfitla_v3/Exercise.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/Exercise.java]: I think this is unused?

File: app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java
Comment (justinnhli) [inline app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:10]: What's the reasoning for price being an String?

File: app/src/main/res/values/strings.xml
Comment (justinnhli) [reply app/src/main/res/values/strings.xml]: This still needs to be fixed, I think.

File: app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java
Comment (wchen-oxy) [reply app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:10]: For $

Comment (justinnhli) [reply app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:10]: For displaying, sure, but the data itself is a double. Only convert it to a string on display.

**Comment by wchen-oxy** [comment]
Okay, @justinnhli Fixed and justified the string.

File: app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java
Comment (wchen-oxy) [reply app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:10]: That and if they accidentally input $ or another symbol then it won't cause an error

Comment (crzeller11) [reply app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:10]: @justinnhli they aren't going to need the number, other than to display it, so is it really necessary that they have it stored as a double?

Comment (justinnhli) [reply app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:10]: In general it's better to follow the type of the data. It's less surprising for someone reading it, and there's always the possibility that in the future it will be used in adding, etc.

**Pull request merged**
